### PR TITLE
Prosenberg/readmeimp

### PR DESCRIPTION
**Pull request by prosenberg1**
Made changes to update the README. Please take a look.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -97,65 +119,56 @@ on a connection provided by the pool will have the same effect as calling:
...
 
 See branch [examples](https://github.com/nuodb/node-nuodb/tree/examples).
 
-## Help
-
-Issues and questions about node-nuodb can be posted on [GitHub][2].
+## Best Practices
 
-## Build and Test
+Any `try` `catch` block in which NuoDB resources are created must be followed by a `finally` block in which NuoDB resources are then cleaned up. Attempting to clean up NuoDB resources in a `try` block can lead to NuoDB resources being created an never cleaned up.
```
Comment (NikTJ777) [inline README.md:132]: Change "NuoDB resources being created an never cleaned up."
-> "NuoDB resources being created `and` never cleaned up."

Comment (prosenberg1) [reply README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -4,16 +4,38 @@ Node.js NAN C++ NuoDB Driver
 
 The node-nuodb add-on for Node.js powers high performance NuoDB Database applications.
 
-Use node-nuodb to connect Node.js 8 and 10 to NuoDB Database.
+Use node-nuodb to connect Node.js a to NuoDB Database.
```
Comment (NikTJ777) [inline README.md:7]: change "a to Nuodb..."
-> "to a NuoDB..."

**Review by NikTJ777 — CHANGES_REQUESTED** [review]
Looks good!

I've caught 1 or 2 typos that should be corrected.

In addition, there are some changes I want in text that you have _not_ changed.
I've added general comments to this review to cover those.

**Comment by NikTJ777** [comment]
In addition to the text you have changed, I have changed I want in text you did _not_ change (yet...) :)

Please change "'After a pool is created the user must initiate it by using the init method:'"
-> "After a pool is created the user must `initialize` it ` ` using the init method:"

Please change: "`This will populate the pool. The pool is unaivable until this completes.`"
-> "This will populate the pool. The pool is `unavailable` until this completes."

Please change: "`Once the pool has been successfully initialized it is ready for use. The user uses:`"
-> "Once the pool has been successfully initialized it is ready for use. The user `calls`:"

Please change: "`Connections that are somehow failing should be returned to the pool`"
-> "Connections that `have failed in any way` should be returned to the pool"

Please change: "`A pool can be shutdown with:`"
-> "A pool `should` be shutdown `by calling`:"

Please change: "This will close all of the pools connections regardless of weither or not they are in use."
-> "This will close all of the pool's connections regardless of `whether` or not they are in use."
or perhaps better -> "This will close all of the `connections in the pool` regardless of `whether` or not they are in use."
(**_Note_**: is it true that Pool.closePool() will close connections that are currently in use?)

**Comment by prosenberg1** [comment]
> Please change: "This will close all of the pools connections regardless of weither or not they are in use." -> "This will close all of the pool's connections regardless of `whether` or not they are in use." or perhaps better -> "This will close all of the `connections in the pool` regardless of `whether` or not they are in use." (**_Note_**: is it true that Pool.closePool() will close connections that are currently in use?)

Yes, Pool.closePool() will close connections that are currently in use.

**Comment by NikTJ777** [comment]
Awesome!

I've found 2 further minor typos:

Please change: "initial size of the pool and the number of connection the pool will aim to maintain, default of 10 is used if no argument is provided."
-> "initial size of the pool and the number of `connections` the pool will aim to maintain, default of 10 is used if no argument is provided."

Please change: "Connections that have been released back to the connection should not be used anymore."
-> "Connections that have been released back to the connection `pool` should not be used anymore."

**Review by NikTJ777 — APPROVED** [review]
Awesome!

This is a major improvement!  :)

I found 2 further minor typos:

Please change: "initial size of the pool and the number of connection the pool will aim to maintain, default of 10 is used if no argument is provided."
-> "initial size of the pool and the number of `connections` the pool will aim to maintain, default of 10 is used if no argument is provided."

Please change: "Connections that have been released back to the connection should not be used anymore."
-> "Connections that have been released back to the connection `pool` should not be used anymore."

File: README.md [hunk 3]
```diff
@@ -4,16 +4,38 @@ Node.js NAN C++ NuoDB Driver
 
 The node-nuodb add-on for Node.js powers high performance NuoDB Database applications.
 
-Use node-nuodb to connect Node.js 8 and 10 to NuoDB Database.
+Use node-nuodb to connect Node.js to a NuoDB Database.
```
Comment (dwhite-nuodb) [inline README.md:7]: Did you remove the version information and not replace it with a specific known version that works?  Or was the point to list the dependency after...

File: README.md [hunk 4]
```diff
@@ -4,16 +4,38 @@ Node.js NAN C++ NuoDB Driver
...
 
 node-nuodb supports basic features of the NuoDB Database and NuoDB C++ client.
 
+## Dependencies
+
+Node 16.16.0+
```
Comment (dwhite-nuodb) [inline README.md:13]: As far as I know this works with 16.13.2, and as far as I know that is the only version we support.  So we should change this number accordingly.

File: README.md [hunk 5]
```diff
@@ -4,16 +4,38 @@ Node.js NAN C++ NuoDB Driver
...
+
 ## Installation
 
-The driver is distributed as Dockers; two variants exist, an ONBUILD
-variant, and a traditional variant. See the documentation for more
```
Comment (dwhite-nuodb) [inline README.md:14]: Was it your intention to remove any information regarding docker?  I think we need to talk to Ariff about that.

**Review by dwhite-nuodb — COMMENTED** [review]
Need to get clarification on docker stuff and I think the Node version needs to be updated.

File: README.md (hunk 3 above)
Comment (prosenberg1) [reply README.md:7]: The point was to just have it in the dependency list.

File: README.md (hunk 4 above)
Comment (prosenberg1) [reply README.md:13]: Happy to make that change.

File: README.md (hunk 5 above)
Comment (prosenberg1) [reply README.md:14]: I wanted to remove any incorrect installation instruction, and as far as I am aware the driver is not distributed as a "Dockers".  I don't see any link between this driver and Docker anymore.

**Pull request merged**